docs/tethys_sdk/paths.rst

```diff
@@ -0,0 +1,268 @@
+* **User Media**: A location were the app can store user uploaded (publicly accessible) information
+
+Read-Only
+---------
+
+* App Public: The location of the app's public directory (i.e. were static files like css and JavaScrip are stored)
+* App Resources: The location where version-controlled app resource files (e.g. configuration files) are stored
+
+
+Getting Paths
+=============
+
+There are several methods for obtaining the paths that are supported in Tethys Platform as follows:
+
+* `Routing Decorators`_
+    * `Controller Decorator`_
+    * `Consumer Decorator`_
+    * `Handler Decorator`_
+* `App Class Methods and Properties`_
+* `Path Functions`_
+
+Routing Decorators
+------------------
+
+The Tethys paths can easily be accessed in controllers, consumers, and handlers by specifying arguments to the various routing decorators. The arguments for each routing decorator is described below:
+
+Controller Decorator
+++++++++++++++++++++
+
+The method for obtaining Tethys paths in controllers is to use the following arguments in ``controller`` decorator:
+
+* ``app_workspace``
+* ``user_workspace``
+* ``app_media``
+* ``user_media``
+* ``app_public``
+* ``app_resouces``
+
+When you specify the argument in the ``controller`` decorator you must also specify the argument in the controller function itself as this is how the path will be passed to the controller function:
+
+.. code-block:: python
+
+    from tethys_sdk.routing import controller
+
+    @controller(app_workspace=True)
+    def my_controller(request, app_workspace):
+       ...
+
+    @controller(user_workspace=True)
+    def my_controller(request, user_workspace):
+       ...
+
+    @controller(app_media=True)
+    def my_controller(request, app_media):
+       ...
+
+    @controller(user_media=True)
+    def my_controller(request, user_media):
+       ...
+
+    @controller(app_public=True)
+    def my_controller(request, app_public):
+       ...
+
+    @controller(app_resources=True)
+    def my_controller(request, app_resources):
+       ...
+
+    @controller(
+        app_workspace=True,
+        user_workspace=True,
+        app_media=True,
+        user_media=True,
+        app_public=True,
+        app_resources=True,
+    )
+    def my_controller(request, app_workspace, user_workspace, app_media, user_media, app_public, app_resources):
+       ...
+
+Note that the argument name in the controller function must match the argument to the ``controller`` decorator since the paths are passed to the controller function as key-word arguments.
+
+To learn more about the ``controller`` decorator, see: :ref:`controller-decorator`
+
+Consumer Decorator
+++++++++++++++++++
+
+To access Tethys paths in consumers, the ``with_paths`` argument to the ``consumer`` decorator should be set to ``True``. The paths will then be accessible as attributes on the class object.
+
+.. code-block:: python
+
+    from channels.generic.websocket import AsyncWebsocketConsumer
```

Comment:
Does this work on a generic `AsyncWebsocketConsumer` or only on the `TethysAsyncWebsocketConsumer`?

Reply:
It works on `AsyncWebsocketConsumer` because the `TethysAshyncWebsocketConsumer` is added as a base by the `consumer` decorator.